File: Advanced Programming Languages/Assignment2/assignment2_section1_cookieClicker.cpp
#include <iostream>
using namespace std;

auto createCookieClicker() {
    int cookies = 0;
    return [cookies]() mutable {
        return ++cookies;
    };
}

int main() {
    auto clickCookie = createCookieClicker();
    cout << clickCookie() << endl;
    cout << clickCookie() << endl;
    cout << clickCookie() << endl;
}

File: Advanced Programming Languages/Assignment2/assignment2_section1_cookieClicker.py
# Based on cookie clicker game. Get cookies when you click.
def create_cookie_clicker():
    cookies = 0

    def click():
        nonlocal cookies
        cookies += 1
        return cookies

    return click


click_cookie = create_cookie_clicker()
print(click_cookie())
print(click_cookie())
print(click_cookie())